### DISCUSSION TITLE
SocketException: Failed to create server socket (OS Error: 以一种访问权限不允许的方式做了一个访问套接字的尝试 , eerrno = 10013), address = 0.0.0.0, port = 53317

### DISCUSSION THREAD

**Discussion by lava114514** [localsend/localsend#935] [L-简体中文]
如题 无法启动服务器

**Comment by xxyzuo** [comment]
同，有解决吗

**Comment by xxyzuo** [comment]
https://gsw945.com/index.php/archives/33/   发现了解决办法

**Reply by numbero** [reply]
正解~
总结一下，需要先在cmd中执行以下命令确认当前windows生效的tcp端口排除范围
`netsh interface ipv4 show excludedportrange protocol=tcp`

之后只需要打开localsend的“设置”，勾选“高级设置”，之后在端口那里填写一个非被系统排除的端口号即可

例如：以下是我的操作系统的排除范围
`协议 tcp 端口排除范围

开始端口    结束端口
----------    --------
      5357        5357
     27339       27339
     49668       49767
     49768       49867
     50000       50059     *
     50160       50259
     50261       50360
     50391       50490
     53266       53365
     55947       56046
     56047       56146
     63711       63810`
因此，我把localsend的端口号改成了53366

**Comment by liuxhit** [comment]
高级设置中改了localsend的端口号能work

**Comment by JasonGrass** [comment]
原因：53317 端口被系统占用了，无法使用。  
最快解决办法，使用管理员身份运行 cmd / powershell，执行如下命令  
net stop winnat  
net start winnat  

能够释放大部分被占用的端口  
查看被占用端口 netsh interface ipv4 show excludedportrange protocol=tcp  （没有标记 * 的就是被占用的）

如果还是不行，
1 可以修改 localsend 使用的端口   
2 或者 将 53317 添加到排除名单之外

参考：  
https://gsw945.com/index.php/archives/33/  
https://www.cnblogs.com/jasongrass/p/17058111.html

**Comment by ruinmi** [comment]
尽管报这个错了，但是localsend还是能正常发送的。

好奇怎么突然会被windows这位保留了呢，之前都用的好好的


📌 工作原理（简化）

当你启动电脑或网卡、VPN、Hyper-V 等网络组件时
Windows 会从高位端口范围（默认 49152–65535）里：

预分配几段连续的端口（每段通常 50 个）

标记为 「excluded」（排除）

这些端口就不会再被 bind() 用于监听服务器套接字

但系统自己可以用它们做出站临时连接（ephemeral port）

所以：

它不是“谁在用”，而是“系统保留起来等会自己用”

也不会出现在 netstat，因为没进程在监听

但你如果试图 bind 这些端口，就会得到 WSAEACCES (10013) 错误
